security/src/main/java/com/fincity/security/service/ClientPasswordPolicyService.java

```diff
@@ -45,29 +45,40 @@ public class ClientPasswordPolicyService extends
 	private CacheService cacheService;
 
 	private final Set<Character> specialCharacters = Set.of('~', '`', '!', '@', '#', '$', '%', '^', '&', '*', '(', ')',
-			'_', '-', '+', '=', '{', '}', '[', ']', '|', '\\', '/', ':', ';', '\"', '\'', '<', '>', ',', '.', '?');
+	        '_', '-', '+', '=', '{', '}', '[', ']', '|', '\\', '/', ':', ';', '\"', '\'', '<', '>', ',', '.', '?');
 
 	@PreAuthorize("hasAuthority('Authorities.Client_Password_Policy_CREATE')")
 	@Override
 	public Mono<ClientPasswordPolicy> create(ClientPasswordPolicy entity) {
 
-		return SecurityContextUtil.getUsersContextAuthentication()
+		return flatMapMono(
+
+		        SecurityContextUtil::getUsersContextAuthentication,
+
+		        ca -> entity.getAppId() == null ? this.dao.checkValidEntity(entity.getClientId())
+		                .switchIfEmpty(securityMessageResourceService.throwMessage(
+		                        msg -> new GenericException(HttpStatus.FORBIDDEN, msg),
+		                        SecurityMessageResourceService.MANDATORY_APP_ID, entity.getClientId()))
+		                : Mono.just(true),
 
-				.flatMap(ca -> {
+		        (ca, validEntry) ->
+				{
 
-					ULong currentUser = ULong.valueOf(ca.getLoggedInFromClientId());
+			        ULong currentUser = ULong.valueOf(ca.getLoggedInFromClientId());
 
-					if (ca.isSystemClient() || currentUser.equals(entity.getClientId()))
-						return super.create(entity);
+			        if (ca.isSystemClient() || currentUser.equals(entity.getClientId()))
```

Comment:
It has to be either system client or entity's client id matches with the context user's client id or entity's client id is managed by the context user's client id.